index.js

```diff
@@ -83,6 +94,116 @@ function attachToBody (options, req, reply, next) {
   })
 }
 
+function getSerializedKeyTokens (key) {
+  const tokens = key.split(/(\[|]|{})/).filter(Boolean)
+  return tokens.some(token => ['[', ']', '{}'].includes(token)) ? tokens : []
+}
+
+function validateSerializedKey (tokens, value) {
+  let isValid = true
+
+  // first token can't be '[', ']', or '{}'
+  if (['[', ']', '{}'].includes(tokens[0])) isValid = false
```

Comment:
Maybe we can extract this Array outside the function, as we can save to instantiate it with every function call.